packages/hash/datastore/postgres/migration/1648547815337_base.ts

```diff
@@ -0,0 +1,613 @@
+      },
+      /**
+       * Note: the entity_types table does not include account_id in its primary key,
+       * some types are shared access across all accounts, and for performance reasons
+       * we do not want to shard the entity_types table by account_id.
+       */
+      account_id: {
+        type: "uuid",
+        notNull: true,
+        references: {
+          name: "accounts",
+        },
+        comment: "",
+        deferrable: true,
+      },
+      created_by_account_id: {
+        type: "uuid",
+        notNull: true,
+        references: {
+          name: "accounts",
+        },
+        comment: "",
+        deferrable: true,
+      },
+      created_at: {
+        type: "timestamp with time zone",
+        notNull: true,
+        comment: "The time at which the first version of this type was created",
+      },
+      metadata_updated_at: {
+        type: "timestamp with time zone",
+        notNull: true,
+        comment: "The time at which the shared type metadata was last updated",
+      },
+      /**
+       * Remaining columns are metadata shared across all versions of an entity
+       */
+      versioned: {
+        type: "boolean",
+        notNull: true,
+        default: true,
+        comment: stripNewLines(`
+          If true, multiple versions of this type may exist in the \`entity_type_versions\`
+          table. For non-versioned types, this column is \`false\` and the type has exactly
+          one corresponding row in entity_type_versions
+        `),
+      },
+      extra: {
+        type: "jsonb",
+      },
+      name: {
+        type: "text",
+        notNull: true,
+      },
+    },
+    {
+      ifNotExists: true,
+    },
+  );
+
+  pgm.addConstraint("entity_types", "entity_type_unique_constraint", {
+    unique: ["account_id", "name"],
+  });
+
+  pgm.createTable(
+    "entity_type_versions",
+    {
+      entity_type_version_id: {
+        type: "uuid",
+        notNull: true,
+        primaryKey: true,
+      },
+      account_id: {
+        type: "uuid",
+        notNull: true,
+        references: {
+          name: "accounts",
+        },
+      },
+      entity_type_id: {
+        type: "uuid",
+        notNull: true,
+        references: {
+          name: "entity_types",
+        },
+      },
+      properties: {
+        type: "jsonb",
+        notNull: true,
+      },
+      updated_by_account_id: {
+        type: "uuid",
+        notNull: true,
+        references: {
+          name: "accounts",
+        },
+      },
+      updated_at: {
+        type: "timestamp with time zone",
+        notNull: true,
+        comment: stripNewLines(`
+          Versioned types are never mutated, so the updated_at time always matches the
+          created_at time. Non-versioned types may be mutatated in-place, and the
+          updated_at column changes when a mutation is made.
+        `),
+      },
+    },
+    {
+      ifNotExists: true,
```

Comment:
I think we should keep the `{ ifNotExists: true }`.
It is very useful for when developing migrations and having a "bad" down migration temporarily which doesn't clean up properly. Having `ifNotExists` means you can still `up` to re-run the `down`. I don't see why we wouldn't just keep it.